src/v/kafka/server/group.cc

```diff
@@ -1926,6 +1926,11 @@ group::begin_tx(cluster::begin_group_tx_request r) {
           "Error \"{}\" on replicating pid:{} fencing batch",
           e.error(),
           r.pid);
+        if (
+          _partition->raft()->is_leader()
+          && _partition->raft()->term() == _term) {
+            co_await _partition->raft()->step_down("group begin_tx failed");
+        }
```

Comment:
surprising, but makes sense. do you think we'd ever want to consider what is contained in `e`, since presumably, not all errors would necessarily result in an ambiguity about the fate of the replication. but perhaps we don't have that granular / precise errors right now.

Reply:
actually the error returned from `replicate` will almost always indicate failed replication, only situation in which the result is ambiguous is the situation in which `raft` is shutting down (we do not wait for replication to finish), so i think we may differentiate here

Reply:
nit: we may rename the `e` as it suggest it is an error whereas in reality is the `result`

Reply:
>only situation in which the result is ambiguous is the situation in which raft is shutting down

wouldn't all forms of timeouts also be ambiguous, too?  now that i think about it, it seems like you'd have to effectively categorize errors such that a class of error was guarnateed to only ever be raised prior to sending messages to replicas in order to have a class of errors that would be non-ambiguous?

Reply:
Right we don't have rigid categorization of errors into ambiguous / non-ambiguous. It's safer to treat all errors as ambiguous.